plugins/SitesManager/API.php

```diff
@@ -278,12 +278,20 @@ public function getSitesIdWithVisits($timestamp = false)
      * For the superUser it returns all the websites in the database.
      *
      * @param bool $fetchAliasUrls
+     * @param false|string $pattern
+     * @param false|int    $limit
      * @return array for each site, an array of information (idsite, name, main_url, etc.)
      */
-    public function getSitesWithAdminAccess($fetchAliasUrls = false)
+    public function getSitesWithAdminAccess($fetchAliasUrls = false, $pattern = false, $limit = false)
```

Comment:
We have to apply `$pattern` and `$limit` already here otherwise we do request eg for 50k websites each the `getSiteUrlsFromId()` call instead of only 20 times when 20 websites are requested. This improves performance a lot.